youtube_dl/extractor/youmaker.py

```diff
@@ -0,0 +1,144 @@
+
+    _TESTS = [
+        {
+            "url": "http://www.youmaker.com/v/71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+            "info_dict": {
+                "id": "71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+                "ext": "mp4",
+                "title": "Как сшить шапочку из трикотажа. Плоский шов двойной иглой.",
+                "description": r"re:(?s)^Привет друзья!\n\nВ этом видео я .* представлена www\.iksonmusic\.com$",
+                "thumbnail": r"re:^https?://.*\.(?:jpg|png)$",
+                "duration": 358,
+                "upload_date": "20190614",
+                "uploader": "user_318f21e00e1f8a6b414f20a654d0f4fc7d2053bc",
+                "timestamp": 1560554895,
+                "channel": "Sewing Ideas",
+                "channel_id": "40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "channel_url": r"re:https?://www.youmaker.com/channel/40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "tags": [
+                    "как сшить детскую шапочку из трикотажа",
+                    "как шить двойной иглой трикотаж",
+                ],
+            },
+            "params": {
+                "skip_download": True,
+            },
+        },
+    ]
+
+    def __init__(self, downloader=None):
+        """Constructor. Receives an optional downloader."""
+        super(YouMakerIE, self).__init__(downloader=downloader)
+        self.__protocol = "https"
+
+    def _fix_url(self, url):
+        if url.startswith("//"):
+            return "%s:%s" % (self.__protocol, url)
+        return url
+
+    @property
+    def _base_url(self):
+        return self._fix_url("//www.youmaker.com")
+
+    @property
+    def _api_url(self):
+        return "%s/v1/api" % self._base_url
+
+    @property
+    def _asset_url(self):
+        # as this url might change in the future
+        # it needs to be extracted from some js magic...
+        return self._fix_url("//vs.youmaker.com/assets")
+
+    def _extract_subtitles(self, system_id=None):
+        subtitles = {}
+
+        if system_id is None:
+            return subtitles
+
+        info = self._download_json(
+            "%s/video/subtitle?systemid=%s" % (self._api_url, system_id),
+            system_id,
+        )
+
+        if info.get("status") != "ok":
+            return subtitles
+
+        for item in info.get("data", []):
+            subtitles.setdefault(item["language_code"], []).append(
+                {"url": "%s/%s" % (self._asset_url, item["url"])}
+            )
+        return subtitles
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        if url.startswith("http://"):
+            self.__protocol = "http"
+        info = self._download_json(
+            "%s/video/metadata/%s" % (self._api_url, video_id), video_id
+        )
+
+        status = info.get("status", "something went wrong")
+        if status != "ok":
+            raise utils.ExtractorError(status, expected=True)
+
+        info = info["data"]
```

Comment:
Now info['data'] needs to be a dict. As an `['id']` is mandatory (as is `['title']`), you could get it here and give up otherwise:
```
        display_id = video_id  # this can be included as the 'display_id' of the result 
        video_id, title = try_get(info, lambda x: (x['data']['id'], x['data']['title'], )
        title = str_or_none(title)
        if video_id is None or not title:
            raise ExtractorError('Unable to extract id/title')
```
Eventually `video_id` and `title` can be used in the result dict.

Reply:
Ok. Now I check for certain mandatory keys, so I can safely use them in the code.